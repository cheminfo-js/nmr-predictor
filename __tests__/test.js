
const nmr = require('..');


const molfile = 'CCc1ccccc1\nJME 2017-11-16 Mon Jun 25 16:33:25 GMT+200 2018\n\n  8  8  0  0  0  0  0  0  0  0999 V2000\n    2.4249    0.7000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0\n    2.4249    2.1000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0\n    1.2124    2.8000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0\n    0.0000    2.1000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0\n    0.0000    0.7000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0\n    1.2124    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0\n    3.6373    2.8000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0\n    4.8497    2.1000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0\n  1  2  1  0  0  0  0\n  2  3  2  0  0  0  0\n  3  4  1  0  0  0  0\n  4  5  2  0  0  0  0\n  5  6  1  0  0  0  0\n  6  1  2  0  0  0  0\n  2  7  1  0  0  0  0\n  7  8  1  0  0  0  0\nM  END\n';

jest.setTimeout(10000);

test('1H prediction from molfile', async function () {
    const ranges = await nmr.spinus(molfile);
    expect(ranges).toMatchSnapshot();
});

test('13C prediction from molfile', async function () {
    await nmr.fetchCarbon();
    const ranges = nmr.carbon(molfile);
    expect(ranges).toMatchSnapshot();
});