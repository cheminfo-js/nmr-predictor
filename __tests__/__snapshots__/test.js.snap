// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1H prediction from molfile 1`] = `
Array [
  Object {
    "atomIDs": Array [
      "8",
    ],
    "atomLabel": "H",
    "delta": 7.162,
    "diaIDs": Array [
      "did@\`@fTfYUn\`HH@GzP\`HeT",
    ],
    "j": Array [
      Object {
        "assignment": Array [
          "9",
        ],
        "coupling": 0.985,
        "diaID": "did@\`@fTfYUn\`HH@GzP\`HeT",
        "distance": 4,
        "multiplicity": "d",
      },
      Object {
        "assignment": Array [
          "10",
        ],
        "coupling": 0.507,
        "diaID": "did@\`@f\\\\bbRaih@J@A~dHBIU@",
        "distance": 5,
        "multiplicity": "d",
      },
      Object {
        "assignment": Array [
          "11",
        ],
        "coupling": 1.293,
        "diaID": "did@\`@fTfUvf\`@h@GzP\`HeT",
        "distance": 4,
        "multiplicity": "d",
      },
      Object {
        "assignment": Array [
          "12",
        ],
        "coupling": 7.758,
        "diaID": "did@\`@f\\\\bbRaih@J@A~dHBIU@",
        "distance": 3,
        "multiplicity": "d",
      },
    ],
    "nbAtoms": 1,
  },
  Object {
    "atomIDs": Array [
      "9",
    ],
    "atomLabel": "H",
    "delta": 7.162,
    "diaIDs": Array [
      "did@\`@fTfYUn\`HH@GzP\`HeT",
    ],
    "j": Array [
      Object {
        "assignment": Array [
          "8",
        ],
        "coupling": 0.985,
        "diaID": "did@\`@fTfYUn\`HH@GzP\`HeT",
        "distance": 4,
        "multiplicity": "d",
      },
      Object {
        "assignment": Array [
          "10",
        ],
        "coupling": 7.758,
        "diaID": "did@\`@f\\\\bbRaih@J@A~dHBIU@",
        "distance": 3,
        "multiplicity": "d",
      },
      Object {
        "assignment": Array [
          "11",
        ],
        "coupling": 1.292,
        "diaID": "did@\`@fTfUvf\`@h@GzP\`HeT",
        "distance": 4,
        "multiplicity": "d",
      },
      Object {
        "assignment": Array [
          "12",
        ],
        "coupling": 0.507,
        "diaID": "did@\`@f\\\\bbRaih@J@A~dHBIU@",
        "distance": 5,
        "multiplicity": "d",
      },
    ],
    "nbAtoms": 1,
  },
  Object {
    "atomIDs": Array [
      "10",
    ],
    "atomLabel": "H",
    "delta": 7.26,
    "diaIDs": Array [
      "did@\`@f\\\\bbRaih@J@A~dHBIU@",
    ],
    "j": Array [
      Object {
        "assignment": Array [
          "8",
        ],
        "coupling": 0.507,
        "diaID": "did@\`@fTfYUn\`HH@GzP\`HeT",
        "distance": 5,
        "multiplicity": "d",
      },
      Object {
        "assignment": Array [
          "9",
        ],
        "coupling": 7.758,
        "diaID": "did@\`@fTfYUn\`HH@GzP\`HeT",
        "distance": 3,
        "multiplicity": "d",
      },
      Object {
        "assignment": Array [
          "11",
        ],
        "coupling": 7.718,
        "diaID": "did@\`@fTfUvf\`@h@GzP\`HeT",
        "distance": 3,
        "multiplicity": "d",
      },
      Object {
        "assignment": Array [
          "12",
        ],
        "coupling": 1.596,
        "diaID": "did@\`@f\\\\bbRaih@J@A~dHBIU@",
        "distance": 4,
        "multiplicity": "d",
      },
    ],
    "nbAtoms": 1,
  },
  Object {
    "atomIDs": Array [
      "11",
    ],
    "atomLabel": "H",
    "delta": 7.196,
    "diaIDs": Array [
      "did@\`@fTfUvf\`@h@GzP\`HeT",
    ],
    "j": Array [
      Object {
        "assignment": Array [
          "8",
        ],
        "coupling": 1.293,
        "diaID": "did@\`@fTfYUn\`HH@GzP\`HeT",
        "distance": 4,
        "multiplicity": "d",
      },
      Object {
        "assignment": Array [
          "9",
        ],
        "coupling": 1.292,
        "diaID": "did@\`@fTfYUn\`HH@GzP\`HeT",
        "distance": 4,
        "multiplicity": "d",
      },
      Object {
        "assignment": Array [
          "10",
        ],
        "coupling": 7.718,
        "diaID": "did@\`@f\\\\bbRaih@J@A~dHBIU@",
        "distance": 3,
        "multiplicity": "d",
      },
      Object {
        "assignment": Array [
          "12",
        ],
        "coupling": 7.718,
        "diaID": "did@\`@f\\\\bbRaih@J@A~dHBIU@",
        "distance": 3,
        "multiplicity": "d",
      },
    ],
    "nbAtoms": 1,
  },
  Object {
    "atomIDs": Array [
      "12",
    ],
    "atomLabel": "H",
    "delta": 7.26,
    "diaIDs": Array [
      "did@\`@f\\\\bbRaih@J@A~dHBIU@",
    ],
    "j": Array [
      Object {
        "assignment": Array [
          "8",
        ],
        "coupling": 7.758,
        "diaID": "did@\`@fTfYUn\`HH@GzP\`HeT",
        "distance": 3,
        "multiplicity": "d",
      },
      Object {
        "assignment": Array [
          "9",
        ],
        "coupling": 0.507,
        "diaID": "did@\`@fTfYUn\`HH@GzP\`HeT",
        "distance": 5,
        "multiplicity": "d",
      },
      Object {
        "assignment": Array [
          "10",
        ],
        "coupling": 1.596,
        "diaID": "did@\`@f\\\\bbRaih@J@A~dHBIU@",
        "distance": 4,
        "multiplicity": "d",
      },
      Object {
        "assignment": Array [
          "11",
        ],
        "coupling": 7.718,
        "diaID": "did@\`@fTfUvf\`@h@GzP\`HeT",
        "distance": 3,
        "multiplicity": "d",
      },
    ],
    "nbAtoms": 1,
  },
  Object {
    "atomIDs": Array [
      "13",
    ],
    "atomLabel": "H",
    "delta": 2.653,
    "diaIDs": Array [
      "did@\`@fTf[Waj@@bJ@_iB@bUP",
    ],
    "j": Array [
      Object {
        "assignment": Array [
          "15",
        ],
        "coupling": 7.392,
        "diaID": "did@\`@fTeYWaj@@@GzP\`HeT",
        "distance": 3,
        "multiplicity": "d",
      },
      Object {
        "assignment": Array [
          "16",
        ],
        "coupling": 7.392,
        "diaID": "did@\`@fTeYWaj@@@GzP\`HeT",
        "distance": 3,
        "multiplicity": "d",
      },
      Object {
        "assignment": Array [
          "17",
        ],
        "coupling": 7.392,
        "diaID": "did@\`@fTeYWaj@@@GzP\`HeT",
        "distance": 3,
        "multiplicity": "d",
      },
    ],
    "nbAtoms": 1,
  },
  Object {
    "atomIDs": Array [
      "14",
    ],
    "atomLabel": "H",
    "delta": 2.653,
    "diaIDs": Array [
      "did@\`@fTf[Waj@@bJ@_iB@bUP",
    ],
    "j": Array [
      Object {
        "assignment": Array [
          "15",
        ],
        "coupling": 7.392,
        "diaID": "did@\`@fTeYWaj@@@GzP\`HeT",
        "distance": 3,
        "multiplicity": "d",
      },
      Object {
        "assignment": Array [
          "16",
        ],
        "coupling": 7.392,
        "diaID": "did@\`@fTeYWaj@@@GzP\`HeT",
        "distance": 3,
        "multiplicity": "d",
      },
      Object {
        "assignment": Array [
          "17",
        ],
        "coupling": 7.392,
        "diaID": "did@\`@fTeYWaj@@@GzP\`HeT",
        "distance": 3,
        "multiplicity": "d",
      },
    ],
    "nbAtoms": 1,
  },
  Object {
    "atomIDs": Array [
      "15",
    ],
    "atomLabel": "H",
    "delta": 0.992,
    "diaIDs": Array [
      "did@\`@fTeYWaj@@@GzP\`HeT",
    ],
    "j": Array [
      Object {
        "assignment": Array [
          "13",
        ],
        "coupling": 7.392,
        "diaID": "did@\`@fTf[Waj@@bJ@_iB@bUP",
        "distance": 3,
        "multiplicity": "d",
      },
      Object {
        "assignment": Array [
          "14",
        ],
        "coupling": 7.392,
        "diaID": "did@\`@fTf[Waj@@bJ@_iB@bUP",
        "distance": 3,
        "multiplicity": "d",
      },
    ],
    "nbAtoms": 1,
  },
  Object {
    "atomIDs": Array [
      "16",
    ],
    "atomLabel": "H",
    "delta": 0.992,
    "diaIDs": Array [
      "did@\`@fTeYWaj@@@GzP\`HeT",
    ],
    "j": Array [
      Object {
        "assignment": Array [
          "13",
        ],
        "coupling": 7.392,
        "diaID": "did@\`@fTf[Waj@@bJ@_iB@bUP",
        "distance": 3,
        "multiplicity": "d",
      },
      Object {
        "assignment": Array [
          "14",
        ],
        "coupling": 7.392,
        "diaID": "did@\`@fTf[Waj@@bJ@_iB@bUP",
        "distance": 3,
        "multiplicity": "d",
      },
    ],
    "nbAtoms": 1,
  },
  Object {
    "atomIDs": Array [
      "17",
    ],
    "atomLabel": "H",
    "delta": 0.992,
    "diaIDs": Array [
      "did@\`@fTeYWaj@@@GzP\`HeT",
    ],
    "j": Array [
      Object {
        "assignment": Array [
          "13",
        ],
        "coupling": 7.392,
        "diaID": "did@\`@fTf[Waj@@bJ@_iB@bUP",
        "distance": 3,
        "multiplicity": "d",
      },
      Object {
        "assignment": Array [
          "14",
        ],
        "coupling": 7.392,
        "diaID": "did@\`@fTf[Waj@@bJ@_iB@bUP",
        "distance": 3,
        "multiplicity": "d",
      },
    ],
    "nbAtoms": 1,
  },
]
`;
